### ISSUE TITLE
Define available callbacks for events used as directives

### ISSUE THREAD

**Issue by paleo229** [monkberry/monkberry#40]
Hi, thanks for this amazing template engine. Here is a suggestion.

I currently use events with this kind of code:

	<input type="text" :onchange="{{ ctrl.changeText }}">

	import Template from './Template.monk';
	let state = {
	  someStateVariable: 'abc',
	  ctrl: {
	    changeText: e => { /* ... */ }
	  }
	}
	view = Monkberry.render(Template, el, {
	  directives: { /* ... */ }
	})
	view.update(state)

I would like to configure available callbacks the same way as for directives:

	import Template from './Template.monk';
	let state = {
	  someStateVariable: 'abc'
	}
	view = Monkberry.render(Template, el, {
	  directives: { /* ... */},
	  callacks: {
	    changeText: e => { /* ... */ }
	  }
	})
	view.update(state)

I prefer to not extend the template class. I just need to define available callbacks, [like `methods` in Vue](https://vuejs.org/v2/guide/events.html#Method-Event-Handlers).

**Comment by elfet** [comment]
So you proposing to add `methods`? Like this:
```js
Monkberry.render(Template, el, {methods:
  callback: () => {...}
});
```

**Comment by paleo229** [comment]
Yes, but I'm unsure that _"methods"_ is the right term. A [method](https://en.wikipedia.org/wiki/Method_(computer_programming)) is in the context of the OOP. But it is precisely in order to avoid OOP that I need this feature. Maybe _"functions"_? Or _"procedures"_?

Or perhaps we can consider it is a way to provide _methods_ of the template (the template is like an instance), but IMO it's not the proper term.

**Comment by elfet** [comment]
`callbacks`?

**Comment by paleo229** [comment]
It is my first proposition. A [callback](https://en.wikipedia.org/wiki/Callback_(computer_programming)).

> In computer programming, a callback is a piece of executable code that is passed as an argument to other code, which is expected to call back (execute) the argument at some convenient time.

We can consider that we pass a list of _callbacks_ as arguments to the code of the template, which we expect that they will be call back at some convenient time. It seems OK.

It depends on which is the syntax in the template. For example, we want to define a callback named `onChangeText`. In the template, which syntax have we to use?

    <input type="text" :onchange="{{ onChangeText }}">

or with a prefix?

    <input type="text" :onchange="{{ controller.onChangeText }}">

Without a prefix, a `callbacks` option seems a good term.

However, if we use a prefix (here `controller`), then I think it would be better to use the same prefix in options:

    Monkberry.render(Template, el, {
      controller: {
        onChangeText: () => {...}
      }
    });

**Comment by elfet** [comment]
Maybe allow to use or not prefixes? 
```js
Monkberry.render(Template, el, {
  callbacks: {
    controller: {
       onChangeText: (event) => {...}
    }
  }
});
```

**Comment by paleo229** [comment]
I'd like that!

**Comment by elfet** [comment]
Will try to implement it in next release.

**Comment by paleo** [comment]
Hello, is there a date scheduled for the release of this feature?

**Comment by antonmedv** [comment]
Hi, 
I haven't time for scheduled work on this project any more. But I have plans on releasing new major version v5, but again – I don't know when. 
If you, or somebody else wants to become a maintainer – your're welcome.
